src/shell.zig

```diff
@@ -319,3 +405,28 @@ test "shell: expand_argv" {
         ),
     );
 }
+
+/// Finds the root of TigerBeetle repo.
+///
+/// Caller is responsible for closing the dir.
+fn discover_project_root() !std.fs.Dir {
+    var current = try std.fs.cwd().openDir(".", .{}); // dup cwd so that the caller can close
```

Comment:
What happened to the git command you just switched `git_root` to using?

Reply:
`git rev-parse --show-top-level` is still the correct way to find git root. But here, I don't actually need git root. I need the root of this particular project. As we control the layout here, we can go for a simpler solution of looking for our specific root file. 

Plus, I want this to be a `Dir` rather than `[]const u8` --- Zig in general is happier when you operate with `fd`s rather than paths.